pkg/apis/sources/v1alpha1/apiserver_types.go

```diff
@@ -132,8 +126,6 @@ type ApiServerResource struct {
 
 	// LabelSelector restricts this source to objects with the selected labels
 	// More info: http://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#label-selectors
-	// Deprecated: Per-resource label selectors will no longer be supported in
-	// v1alpha2, please use Spec.LabelSelector.
 	LabelSelector metav1.LabelSelector `json:"labelSelector"`
 
 	// ControllerSelector restricts this source to objects with a controlling owner reference of the specified kind.
```

Comment:
I agree with removing `ControllerSelector` (redundant). Not so sure about deprecating `Controller` as it has a real use case. k8s sample controller does it sightly differently (see https://github.com/kubernetes/sample-controller/blob/041853156b367ba768df5cda747e653dd98a7127/controller.go#L370) by getting the controlling owner and filtering by GVK.

Reply:
I am filtering by gvk, but the owner gvk comes from spec. If you need both resources that are owned and unowned, it would be two resources. 

I think next pass is to mt the source in a namespace for the same service account.